### PR TITLE
lambda conversion

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/content/onlineservice/sopcast_content_handler.cc [hunk 1]
```diff
@@ -144,18 +144,18 @@ std::shared_ptr<CdsObject> SopCastContentHandler::getObject(const std::string& g
...
+        if (temp == "wmv")
+            return "video/sopcast-x-ms-wmv";
+        if (temp == "mp3")
+            return "audio/sopcast-mpeg";
+        if (temp == "wma")
+            return "audio/sopcast-x-ms-wma";
+        return "application/sopcast-stream";
+    }();
+    if (mt == "application/sopcast-stream")
         log_warning("Could not determine mimetype for SopCast channel (stream_type: {})", temp.c_str());
```
Comment (KarlStraussberger) [inline src/content/onlineservice/sopcast_content_handler.cc:158]: Why don't you move that warning into the lambda?

File: src/content/onlineservice/sopcast_content_handler.cc [hunk 2]
```diff
@@ -144,18 +144,18 @@ std::shared_ptr<CdsObject> SopCastContentHandler::getObject(const std::string& g
     // I wish they had a mimetype setting
     //auto mappings = config->getDictionaryOption(CFG_IMPORT_MAPPINGS_EXTENSION_TO_MIMETYPE_LIST);
```
Comment (KarlStraussberger) [inline src/content/onlineservice/sopcast_content_handler.cc:145]: I suggest to create a new config dictionary `CFG_IMPORT_SOPCAST_MIMETYE_LIST` defaulting to the current values and some string with the default value.

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]

File: src/content/onlineservice/sopcast_content_handler.cc (hunk 2 above)
Comment (neheb) [reply src/content/onlineservice/sopcast_content_handler.cc:145]: That's beyond the scope here.

File: src/content/onlineservice/sopcast_content_handler.cc (hunk 1 above)
Comment (neheb) [reply src/content/onlineservice/sopcast_content_handler.cc]: Good catch.

**Review by KarlStraussberger — APPROVED** [review]

**Pull request merged**